WordPressKit/Time-based data/SummaryStatsType.swift

```diff
@@ -0,0 +1,127 @@
+        // is the `period`, second is `views`, etc.
+
+        guard
+            let periodIndex = fieldsArray.firstIndex(of: "period"),
+            let viewsIndex = fieldsArray.firstIndex(of: "views"),
+            let visitorsIndex = fieldsArray.firstIndex(of: "visitors"),
+            let likesIndex = fieldsArray.firstIndex(of: "likes"),
+            let commentsIndex = fieldsArray.firstIndex(of: "comments")
+            else {
+                return nil
+        }
+
+        self.period = period
+        self.periodEndDate = date
+        self.summaryData = data.compactMap { StatsSummaryData(dataArray: $0,
+                                                              period: period,
+                                                              periodIndex: periodIndex,
+                                                              viewsIndex: viewsIndex,
+                                                              visitorsIndex: visitorsIndex,
+                                                              likesIndex: likesIndex,
+                                                              commentsIndex: commentsIndex) }
+    }
+}
+
+private extension StatsSummaryData {
+    init?(dataArray: [Any],
+          period: StatsPeriodUnit,
+          periodIndex: Int,
+          viewsIndex: Int,
+          visitorsIndex: Int,
+          likesIndex: Int,
+          commentsIndex: Int) {
+        guard
+            let periodString = dataArray[periodIndex] as? String,
+            let periodStart = type(of: self).parsedDate(from: periodString, for: period),
+            let viewsCount = dataArray[viewsIndex] as? Int,
+            let visitorsCount = dataArray[visitorsIndex] as? Int,
+            let likesCount = dataArray[likesIndex] as? Int,
+            let commentsCount = dataArray[commentsIndex] as? Int
+            else {
+                return nil
+        }
+
+        self.period = period
+        self.periodStartDate = periodStart
+        self.viewsCount = viewsCount
+        self.visitorsCount = visitorsCount
+        self.likesCount = likesCount
+        self.commentsCount = commentsCount
+    }
+
+    static func parsedDate(from dateString: String, for period: StatsPeriodUnit) -> Date? {
+        switch period {
+        case .week:
+            return self.weeksDateFormatter.date(from: dateString)
+        case .day, .month, .year:
+            return self.regularDateFormatter.date(from: dateString)
+        }
+    }
+
+    static var regularDateFormatter: DateFormatter {
+        let df = DateFormatter()
+        df.locale = Locale(identifier: "en_US_POS")
+        df.dateFormat = "yyyy-MM-dd"
+        return df
+    }
+
+    // We have our own handrolled date format for data broken up on week basis.
+    // Example dates in this format are `2019W02W18` or `2019W02W11`.
+    // The structure is rougly `aaaaWbbWcc`, where:
```

Comment:
Minor typo - `rougly` = `roughly`. I know, you _hate_ when I actually read the comments. 😉 

Oh, yea, thanks for adding these descriptive comments!

Reply:
:)